scripts/interopTestClient.ts

```diff
@@ -0,0 +1,256 @@
+}
+
+interface HistogramVdafObject {
+    type: "Prio3Aes128Histogram",
+    buckets: string[],
+}
+
+type VdafObject = CountVdafObject | SumVdafObject | HistogramVdafObject;
+
+type Measurement = number | string | string[];
+
+interface UploadRequest {
+    taskId: string,
+    leader: string,
+    helper: string,
+    vdaf: VdafObject,
+    measurement: Measurement,
+    nonceTime?: number,
+    minBatchDuration: number,
+}
+
+function sanitizeRequest(rawBody: unknown): UploadRequest {
+    if (typeof rawBody !== "object" || Array.isArray(rawBody) || rawBody === null) {
+        throw new Error("JSON body is not an object");
+    }
+
+    if (!("taskId" in rawBody)) {
+        throw new Error("Task ID is missing");
+    }
+    if (!("leader" in rawBody)) {
+        throw new Error("Leader endpoint URL is missing");
+    }
+    if (!("helper" in rawBody)) {
+        throw new Error("Helper endpoint URL is missing");
+    }
+    if (!("vdaf" in rawBody)) {
+        throw new Error("VDAF definition is missing");
+    }
+    if (!("measurement" in rawBody)) {
+        throw new Error("Measurement is missing");
+    }
+    if (!("minBatchDuration" in rawBody)) {
+        throw new Error("Minimum batch duration is missing");
+    }
+
+    const body = rawBody as {
+        taskId: unknown,
+        leader: unknown,
+        helper: unknown,
+        vdaf: unknown,
+        measurement: unknown,
+        nonceTime?: unknown,
+        minBatchDuration: unknown,
+    };
+
+    if (typeof body.taskId !== "string") {
+        throw new Error("Task ID is not a string");
+    }
+    if (typeof body.leader !== "string") {
+        throw new Error("Leader endpoint URL is not a string");
+    }
+    if (typeof body.helper !== "string") {
+        throw new Error("Helper endpoint URL is not a string");
+    }
+    if (typeof body.vdaf !== "object" || Array.isArray(body.vdaf) || body.vdaf === null) {
+        throw new Error("VDAF definition is not an object");
+    }
+    if (typeof body.measurement !== "number" && typeof body.measurement !== "string" && !Array.isArray(body.measurement)) {
+        throw new Error("Measurement is not a number, string, or array");
+    }
+    if (body.nonceTime !== undefined && typeof body.nonceTime !== "number") {
+        throw new Error("Nonce timestamp is not a number");
+    }
+    if (typeof body.minBatchDuration !== "number") {
+        throw new Error("Minimum batch duration is not a number");
+    }
+
+    if (!("type" in body.vdaf)) {
+        throw new Error("VDAF type is missing");
+    }
+
+    const vdaf = body.vdaf as {
+        type: unknown,
+        bits?: unknown,
+        buckets?: unknown,
+    };
+
+    if (typeof vdaf.type !== "string") {
+        throw new Error("VDAF type is not a string");
+    }
+
+    let vdafObject: VdafObject;
+    let measurement: Measurement;
+
+    switch (vdaf.type) {
+        case "Prio3Aes128Count":
+            if (body.measurement !== 0 && body.measurement !== 1) {
```

Comment:
Do we need to potentially convert the measurement from a string before this check? Since `'0'` and `'1'` pass the typechecking above, I think so -- I would expect `'0'` (string) to be treated the same as `0` (numeric), but it would fail this check.

Reply:
Based on the current spec text, that's not required, but I'll add a conversion to be more permissive, plus we may change that part of the spec to use strings soon anyhow.

Reply:
Thanks. Given the spec text, I would have accepted a strict interpretation of the spec (i.e. not accepting stringified numbers here), but I also definitely support the spec changing to represent all numbers (or at least all numbers used to represent part of a measurement) as string values -- so being more permissive SGTM.